benches/distribution_statistic.rs

```diff
@@ -0,0 +1,73 @@
+use criterion::{criterion_group, criterion_main, BatchSize, Criterion};
+
+use rand::distributions::Distribution;
+use rand::distributions::Uniform;
+use rand::rngs::SmallRng;
+use rand::seq::SliceRandom;
+use rand::SeedableRng;
+
+use vector::event::metric::Sample;
+use vector::sinks::util::statistic::DistributionStatistic;
+
+fn generate_samples(mut size: u32, max_bin_count: u32) -> Vec<Sample> {
```

Comment:
Do we have any idea if these generated samples are at all representative of actual data?

Reply:
Depend on what kind of data this is observing, `samples` could be more accurately represented by a normal distribution.

I don't know how the real data look, so there are two set of sample that hopefully represent all possible data:
- Small bin counts for more lightweight data, or when the data is too spread-out.
- Large bin counts.

The sample values don't really matter in this case so I didn't randomize them.